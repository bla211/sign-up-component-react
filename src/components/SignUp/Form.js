import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Input from '../Common/Input';
import Button from '../Common/Button';

class Form extends Component {
    render(){
        const { signUpState } = this.props;
        const { updateEmail } = this.props;
        const { updateFirstName } = this.props;
        const { updateLastName } = this.props;
        const { toggleAgreesToDisclaimer } = this.props;
        const { updateFormState } = this.props;
        
        let input, button, h1, h2, h3, h4, disclaimer;
        if(signUpState.formState === 'email'){
            h1 = <h1>Join the list</h1>;
            h2 = <h2>
                    <span class="line">Sign up for</span>&nbsp;<span class="line">the TLC newsletter</span>
                </h2>;
            input = <Input placeholder="enter email address" type="email" className="full-width" val={ signUpState.userInfo.email } handleChange={ updateEmail }/>;
            button = <Button title="next" onclick={ updateFormState } data="name"/>
            disclaimer = <div id="disclaimer">
                            <label class="checkbox-wrapper">
                                <input type="checkbox" value={ signUpState.userInfo.isAgreeToDisclaimer } onChange={ () => toggleAgreesToDisclaimer() }/>
                                <span class="checkmark"></span>
                            </label>
                            <p class="disclaimer-text">
                                I agree to receive information from Discovery Communications in accordance with the following Privacy Policy
                            </p>
                        </div>
        }
        else if(signUpState.formState === 'name'){
            h1 = <h1>Join the list</h1>;
            h2 = <h2>
                    <span class="line">Almost done! Please enter</span>&nbsp;<span class="line">your first and last name</span>
                </h2>;
            input = <div id="input-wrapper">
                        <Input placeholder="First Name" className="half-width" val={ signUpState.userInfo.name.first } handleChange={ updateFirstName }/>
                        <Input placeholder="Last Name" className="half-width" val={ signUpState.userInfo.name.last } handleChange={ updateLastName }/>
                    </div>;
            button = <Button title="sign up" onclick={ updateFormState } data="congrats"/>
        }
        else if(signUpState.formState === 'congrats'){
            h1 = <h1>Congratulations!</h1>;
            h3 = <h3>
                    <span class="line">Thank you for</span>&nbsp;<span class="line">signing up!</span>
                </h3>;
            h4 = <h4>Look out for the latest news on your favorite shows.</h4>
        }

        return(
            <div id="sign-up-form">
                { h1 }
                { h2 }
                { h3 }
                { h4 }
                { input }
                { button }
                { disclaimer }        
            </div>
        )
    }

}

Form.propTypes = {
    updateEmail: PropTypes.string,
    updateFirstName: PropTypes.string,
    updateLastName: PropTypes.string,
    updateFormState: PropTypes.string,
    signUpState: PropTypes.object,
    toggleAgreesToDisclaimer: PropTypes.string
  };

export default Form;