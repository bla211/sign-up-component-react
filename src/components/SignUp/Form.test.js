import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Form from './Form';

configure({ adapter: new Adapter() });

describe('Test Form component', () => {
    test('Form renders properly when passed props', () => {
        const signUpState = {
            formState: 'email',
            showErrors: false,
            userInfo: {
                email: 'test@test.com',
                name: {
                    first: '',
                    last: ''
                },
                isAgreesToDisclaimer: true
            }
        }
        const wrapper = shallow(<Form signUpState={signUpState}/>);
        expect(wrapper).toMatchSnapshot()
    });

    test('Form renders properly when passed props', () => {
        const signUpState = {
            formState: 'name',
            showErrors: false,
            userInfo: {
                email: 'test@test.com',
                name: {
                    first: 'Test',
                    last: 'Test'
                },
                isAgreesToDisclaimer: true
            }
        }
        const wrapper = shallow(<Form signUpState={signUpState}/>);
        expect(wrapper).toMatchSnapshot()
    });

    test('Form renders properly when passed props', () => {
        const signUpState = {
            formState: 'congrats',
            showErrors: false,
            userInfo: {
                email: 'test@test.com',
                name: {
                    first: 'Test',
                    last: 'Test'
                },
                isAgreesToDisclaimer: true
            }
        }
        const wrapper = shallow(<Form signUpState={signUpState}/>);
        expect(wrapper).toMatchSnapshot()
    });
});