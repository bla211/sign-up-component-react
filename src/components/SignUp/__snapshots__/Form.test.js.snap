// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Form component Form renders properly when passed props 1`] = `
<div
  id="sign-up-form-wrapper"
>
  <div
    className="email"
    id="sign-up-form"
  >
    <Heading
      headingType="h1"
      message={
        Array [
          "Join the list",
        ]
      }
    />
    <div
      id="sign-up-form-inner"
    >
      <Heading
        headingType="h2"
        message={
          Array [
            "Sign up for",
            "the TLC newsletter",
          ]
        }
      />
      <Input
        className={
          Array [
            "full-width",
            undefined,
          ]
        }
        data="name"
        inputName="email"
        isError={false}
        placeholder="enter email address"
        type="text"
        val="test@test.com"
      />
      <Button
        data="name"
        isError={false}
        title="next"
      />
      <Disclaimer
        disclaimerMessage="I agree to receive information from Discovery Communications in accordance with the following Privacy Policy"
        isChecked={true}
      />
    </div>
  </div>
</div>
`;

exports[`Test Form component Form renders properly when passed props 2`] = `
<div
  id="sign-up-form-wrapper"
>
  <div
    className="name"
    id="sign-up-form"
  >
    <Heading
      headingType="h1"
      message={
        Array [
          "Join the list",
        ]
      }
    />
    <div
      id="sign-up-form-inner"
    >
      <Heading
        headingType="h2"
        message={
          Array [
            "Almost done! Please enter",
            "your first and last name",
          ]
        }
      />
      <div
        id="input-wrapper"
      >
        <Input
          autoFocus={true}
          className={
            Array [
              "half-width",
              undefined,
            ]
          }
          data="name"
          inputName="given-name"
          isError={false}
          placeholder="First Name"
          val="Test"
        />
        <Input
          className={
            Array [
              "half-width",
              undefined,
            ]
          }
          data="congrats"
          inputName="family-name"
          isError={false}
          placeholder="Last Name"
          val="Test"
        />
      </div>
      <Button
        data="congrats"
        isError={false}
        title="sign up"
      />
    </div>
  </div>
</div>
`;

exports[`Test Form component Form renders properly when passed props 3`] = `
<div
  id="sign-up-form-wrapper"
>
  <div
    className="congrats"
    id="sign-up-form"
  >
    <Heading
      headingType="h1"
      message={
        Array [
          "Congratulations!",
        ]
      }
    />
    <div
      id="sign-up-form-inner"
    >
      <Heading
        headingType="h3"
        message={
          Array [
            "Thank you for",
            "signing up!",
          ]
        }
      />
      <Heading
        headingType="h4"
        message={
          Array [
            "Look out for the latest news on your favorite shows.",
          ]
        }
      />
    </div>
  </div>
</div>
`;
